troy-schema/src/main/scala/troy/schema/SchemaEngine.scala

```diff
@@ -117,10 +130,61 @@ case class SchemaEngineImpl(schema: Schema, context: Option[KeyspaceName]) exten
       )).map(_.flatten)
     }
 
+  private def extractVariableTypes(s: UpdateStatement): Result[Seq[DataType]] =
+    schema.getTable(s.tableName).flatMap { table =>
+      V.merge(Seq(
+        extractVariablesFomSet(table, s.set),
+        extractVariablesFromUpdateParam(table, s.using),
```

Comment:
Wrong order. Variables in `USING` should appear before variables in `SET`.
Please first write test cases to expose the problem.